feat: improve redirect action documentation (#1638)
ngrok/ngrok-docs: traffic-policy/actions/redirect.mdx

## Behavior

> The Redirect Action modifies the request URL and returns an HTTP redirect (`3xx`) response. It performs this transformation in sequence:

Reviewer: nitpick for clarity

```suggestion
The Redirect Action modifies the request URL and returns an HTTP redirect (`3xx`) response. 
It performs the following transformation in sequence:
```

1. Evaluate any CEL interpolation
2. Apply the regular expression match (`from`)
3. Replace the matched text or URL with the `to` value
4. Send the redirect response

If the rewritten URL is identical to the original, ngrok skips the redirect.

> ### CEL Interpolation

Reviewer: We should be using sentence case everywhere for headers

```suggestion
### CEL interpolation
```

The following example redirects any request that contains `/api/v1/` to `/api/v2/`. The redirect is applied to the entire URL, not just the path.

```yaml
from: /api/v1/
to: /api/v2/
```

Given the above, `https://example.com/foobar/api/v1/users` would be redirected to `https://example.com/foobar/api/v2/users`.

#### Anchoring / Path Prefix

Anchoring with `^/api/v1/` will not match paths that include the full scheme. To match only the path, use one of the following methods:

<CodeGroup>

```yaml Regular Expressions
# Anchor the full URL, capture authority, and rewrite the path segment.
from: ^((?:https?)://[^/]+)/api/v1/(.*)$
to: $1/api/v2/$2
```

```yaml CEL Interpolation
# Compose anchors with CEL to bind the scheme and authority before matching.
from: ^${req.url.scheme}://${req.url.authority}/api/v1/
to: ${req.url.scheme}://${req.url.authority}/api/v2/
```

```yaml Phase Expressions
# Constrain by path in expressions, then use a relative replacement.
on_http_request:
  - expressions:
      - req.url.path.startsWith('/api/v1/')
    actions:
      - type: redirect
        config:
		  from: /api/v1/
          to: /api/v2/
```

</CodeGroup>

### Replacement Behavior

> The `to` value replaces what was matched by `from`. The replacement is **`relative to the from match`**. When `from` is omitted, the redirect action replaces the entire URL. Capture groups (`$1`, `$2`, etc.) can be used to insert matched text into the replacement. 

Reviewer: - the **` situation seems odd to me
- best practice: avoid latinisms like etc., i.e., e.g., and so on

```suggestion
The `to` value replaces what was matched by `from`. 
The replacement is relative to the `from` match. 
When `from` is omitted, the redirect action replaces the entire URL. 
Capture groups (such as `$1`, `$2`) can be used to insert matched text into the replacement. 
```

#### Relative vs. absolute replacements

The `to` value is applied relative to the `from` match. When `from` matches part of the URL, only that segment is replaced. When `from` is omitted, ngrok treats the entire URL as matched and replaces it absolutely.

**Relative example**

```yaml
from: /old
to: https://new.example.com/blog
```

Given the above, `https://example.com/old/foo` would become `https://example.com/https://new.example.com/blog/foo`.

**Absolute example**

When a partial match exists, the replacement is relative; when no `from` match is defined, it is absolute:

```yaml
to: https://new.example.com/blog
```

Given the above, any request to `https://example.com` would be redirected to `https://new.example.com/blog`.

**Redirecting to another host**

To redirect cleanly to a new host, match the entire URL:

```yaml
from: ^https?://[^/]+/old/(.*)$
to: https://new.example.com/new/$1
```

Given the above, `https://example.com/old/foo` would be redirected to `https://new.example.com/new/foo`.

**Note:** If the rewritten URL equals the original, ngrok skips the redirect.

> ### Redirect Response

Reviewer: ```suggestion
### Redirect response
```
